### ISSUE TITLE
Robustness of GDN

### ISSUE THREAD

**Issue by peerschuett** [wagner-d/TimeSeAD#13]
Hi,

In your paper, you mention GDN as one of the best approaches for TSAD. I did some experiments with it too and noticed strong fluctuations in its performance between runs. Did you experience anything similar? In #11 , you mentioned that the results in the paper are "cross-validation results by doing grid search." What do you mean by that exactly? Doesn't this just mean you did a hyperparameter optimization and chose the best parameters?

Best regards,  
Peer

**Comment by batman-nair** [comment]
Could you please add a bit more detail on what changes you made between runs? Are they multiple seeds or different configurations of GDN?
Appendix D of our paper has a detailed explanation of the evaluation procedure. A short overview, we train the model on the normal data and split the abnormal data into multiple folds. One of the folds is used to tune the detector while others are used to evaluate the model.

**Comment by peerschuett** [comment]
Hi,

Thanks for your answer! My initial questions weren't worded well; sorry about that! I'll try again:

Regarding the seed: I executed the original script of GDN from their code release (https://github.com/d-ailin/GDN) with a fixed random seed. I assumed the results would be the same across multiple runs (with the same seed and hyperparameters), but that wasn't the case. I realize now that I shouldn’t ask this question here, but in the original repo.

Regarding cross-validation: Thanks for pointing me to Appendix D! That answered my questions. I had a misunderstanding about cross-validation, which led to my confusion.
